Stage 1: Basic Design, Simple Backend, and User homepage

BlazinSamurai opened the pull request:
### Description

Paper Planes is going to be a feature rich site with elegant design and transition/animations to help a solo traveler plan for their next adventure. 

### Landing Page

**Includes:** 

- Navigation Bar: Includes a logo, which redirect user to the landing page, a login link and a signup link.
- Header Title Text: A little catchy slogan.
- Globe Section: _This is where I connect my outside API, MapBox_. MapBox allows you to have a globe display to see the world any way you choose to see. The landing page globe allows users just to zoom in and out of areas.
- Steps Sections: This section is just a visual representation of how to sign up for a account and what the "Home Page" looks like once the user signs up.

### Signup Page

- This page allows the user to sign up for a free account. Kept it similar to 'WTWR' project.
- Requirements: Username, email, password, profile picture

### Login Page

- Once a user has successfully signed up for a account they can then log in through this page.
- Requirements: Email, and Password

### Home Page

- If a successful signup or login the user is redirected to their "Home Page". 

**Includes:**

- Side Bar: This section has the logo, the user's profile picture, the user's username and a logout button at the bottom.
- Main Section: Title, a "Create A Trip" button, and another Mapbox globe. The user can create a new trip itinerary by clicking the button described above, which open a "New Trip" popup.
- "New Trip" popup: Allows the user to create a visual representation of upcoming trips. 
- "New Trip" requirements: Destination, Trip Name, Start and End Date 
 

**_P.S_**  
- I wasn't able to implement all of my designs I wanted for this site at this stage due to the time constraint and wanted to turn in the core features and functionality. 
- As of this stage the user cannot save their trip as that involves more backend programming which is not necessary for this stage. 
- Also at this stage the globe API is more of just a visual aspect. Later on I want the user to be able to logged in and they will be allowed to click a location on the globe and open up the "New Trip" popup but I also haven't implemented this feature since it is not necessary and would take more time to implement.

lipasofra-tripleten commented:
Hello! Thanks for sending your project for review, it looks great! I like the idea and your implementation. The project looks very good in general, but will require some refactoring.

# Summary:
### Performance Criteria

#### Markup and JSX
- [ ] <!--7.4--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the overflow: hidden declaration.
			-  Page is not displaying correctly for intermediate resolutions https://prnt.sc/glfdYOapEvV9
- [x] <!--7.26--> Elements specified in the UI toolkit are used correctly (e.g., buttons, forms, text elements, etc).
- [x] <!--3.63--> Classes are named according to BEM specifications.
- [x] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--3.63--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--3.63--> Infrastructural project files are created using Vite or CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [ ] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [ ] A `fonts` folder.
- [ ] <!--3.63--> There are no linting errors or warnings.
		- Eslint rules are not correct, please verify it: https://prnt.sc/I73BNEK6YXhQ

#### React and connecting a third-party API
- Routes:
  - [ ] <!--3.63--> There are two separate routes: the root route (`/`) and one other (e.g., `/profile` or `/about`).
  - [ ] <!--3.63--> Navigation between pages and links to external resources is working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [x] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`.
- [x] <!--3.63--> No third-party JavaScript libraries are used, unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [ ] <!--1.25--> The icons are exported in SVG format.
		- some icons are .png and jpg
- [x] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [ ] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with `addEventListener` are removed when the component is unmounted.
- [ ] <!--1.25--> API error handling:
  - [ ] The user receives a message in the case of an error.
  - errors are being consoled, but user cannot see that. You should inform the user if there's an error
- [x] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
- [x] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed using the cross button, overlay, or ESC key.

### Recommendations

#### Markup and JSX
- [ ] <!--0.83--> System fonts are connected as alternatives to each of your fonts.
		- no fallback font are being used
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [x] <!--0.83--> Raster and vector images have been optimized.
		- the cloudBody image could be optimized

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [x] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks.
- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and clearly structured. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, when a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, and the indentation hierarchy is respected.

## Number of points: 73

## Additional comments:
- the PR must only contain frontend code, please remove the backend code from this PR.
- According to project instructions: - Simulate some backend responses to allow the reviewer full access to all pages and design elements. https://prnt.sc/9Eob9W93Fpim I should be able to test the frontend part of the project without the backend part - in this moment I'm not able to simulate login or signup flow.

BlazinSamurai commented:
Hello @lipasofra-tripleten, thank you for the review. I might need a little more clarification about the routes section of the review before I start the refactoring process.

The requirement for the project needs two routes, 1 for the root('/') and one other route, as stated above and my project includes 4 routes. 
- i.e) Root/"Landing Page" ('/'), Sign up ('/signup'), Login ('/login'), and a Home Page ('/homepage') route

My question is why didn't I get a completion mark for routes section when I have 4 and the requirement is only 2? Is it the way these routes were coded that you didn't like or does having a signup and login route not count as "other route"? 

Also I noticed you didn't mark the "Navigation between pages . . ." bullet under the same section as not completed either. Was there something wrong when you clicked the buttons?

I am confused and unsure how to fix these issues. If you can provide a little more clarification that would be greatly appreciated. Thank you

BlazinSamurai commented:
Hello @lipasofra-tripleten,

I've fixed most uncheck box to the best of my ability.

**What I have added:**

### **Performance Criteria**
### Markup and JSX

- I have adapted my project to adjust the layout to multiple screen. I have adjusted the landing page, login/sign up pages, home page, modals, and all their respected components to adjust to various size ranging from 1440px to 320px.
- I changed the vendor folder currently holding my fonts to its own directory called "fonts" to make easier to located.
- Added linter and fixed linter errors. Kept linter setting basic.

### React and connecting a third-party API

- I have four routes: Root/"Landing Page" ('/'), Sign up ('/signup'), Login ('/login'), and a Home Page ('/homepage') route. 

### **Best Practices**
### React

- I have added a temporary user state variable to allow signup and login inside App component.
- Users get alerted when a visual que when password or username/email is incorrect.

### **Recommendations**
### Markup and JSX

- Added two fallback fonts for all components that use imported font.

lipasofra-tripleten commented:
> Hello @lipasofra-tripleten,
> 
> I've fixed most uncheck box to the best of my ability.
> 
> **What I have added:**
> 
> ### **Performance Criteria**
> ### Markup and JSX
> * I have adapted my project to adjust the layout to multiple screen. I have adjusted the landing page, login/sign up pages, home page, modals, and all their respected components to adjust to various size ranging from 1440px to 320px.
> * I changed the vendor folder currently holding my fonts to its own directory called "fonts" to make easier to located.
> * Added linter and fixed linter errors. Kept linter setting basic.
> 
> ### React and connecting a third-party API
> * I have four routes: Root/"Landing Page" ('/'), Sign up ('/signup'), Login ('/login'), and a Home Page ('/homepage') route.
> 
> ### **Best Practices**
> ### React
> * I have added a temporary user state variable to allow signup and login inside App component.
> * Users get alerted when a visual que when password or username/email is incorrect.
> 
> ### **Recommendations**
> ### Markup and JSX
> * Added two fallback fonts for all components that use imported font.

Hi! sorry, haven't notice you answered, next time you could write me in the hub I'm as Lina Soler. I'm going to review your project now and give feedback in other comment.
As commented previously: this PR should only have frontend code, please remove all backend code

lipasofra-tripleten commented:
Hello! Thanks for the corrections. Excellent work!🚀 I am glad to say that your project has been accepted ✨ Please read additional comments at the end of the review summary.

# Summary:
### Performance Criteria

#### Markup and JSX
- [x] <!--7.4--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--7.26--> Elements specified in the UI toolkit are used correctly (e.g., buttons, forms, text elements, etc).
- [x] <!--3.63--> Classes are named according to BEM specifications.
- [ ] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
			- **This could be better a list**: https://prnt.sc/KdGwEmqFY48H
- [x] <!--3.63--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--3.63--> Infrastructural project files are created using Vite or CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [x] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--3.63--> There are no linting errors or warnings.

#### React and connecting a third-party API
- Routes:
  - [x] <!--3.63--> There are two separate routes: the root route (`/`) and one other (e.g., `/profile` or `/about`).
  - [x] <!--3.63--> Navigation between pages and links to external resources is working correctly. There are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and there are no hidden blocks.
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [x] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`.
- [x] <!--3.63--> No third-party JavaScript libraries are used, unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [x] <!--1.25--> The icons are exported in SVG format.
- [x] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [x] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with `addEventListener` are removed when the component is unmounted.
- [x] <!--1.25--> API error handling:
  - [x] The user receives a message in the case of an error.
- [x] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
- [x] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed using the cross button, overlay, or ESC key.

### Recommendations

#### Markup and JSX
- [x] <!--0.83--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [x] <!--0.83--> Raster and vector images have been optimized.

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [x] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks.
- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and clearly structured. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, when a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, and the indentation hierarchy is respected.

## Number of points: 96

## Additional comments:
- This PR should only have the frontend code, please all backend code
- empty files should be removed, for example the Footer file
- when trying to enter to the homepage being logged out (to test the protectedRoute) I'm redirected to the login page but it is blank: https://prnt.sc/czxtoo_liD81
